go-controller/pkg/node/healthcheck.go

```diff
@@ -93,29 +94,113 @@ func countLocalEndpoints(ep *kapi.Endpoints, nodeName string) int {
 	return num
 }
 
-// check for OVS internal ports without any ofport assigned, they are stale ports that must be deleted
-func checkForStaleOVSInterfaces(stopChan chan struct{}) {
-	for {
-		select {
-		case <-time.After(60 * time.Second):
-			stdout, _, err := util.RunOVSVsctl("--data=bare", "--no-headings", "--columns=name", "find",
-				"interface", "ofport=-1")
+// checkForStaleOVSInternalPorts checks for OVS internal ports without any ofport assigned,
+// they are stale ports that must be deleted
+func checkForStaleOVSInternalPorts() {
+	stdout, _, err := util.RunOVSVsctl("--data=bare", "--no-headings", "--columns=name", "find",
+		"interface", "ofport=-1")
+	if err != nil {
+		klog.Errorf("Failed to list OVS interfaces with ofport set to -1")
+		return
+	}
+	if len(stdout) == 0 {
+		return
+	}
+
+	values := strings.Split(stdout, "\n\n")
+	for _, val := range values {
+		klog.Warningf("Found stale interface %s, so deleting it", val)
+		_, stderr, err := util.RunOVSVsctl("--if-exists", "--with-iface", "del-port", val)
+		if err != nil {
+			klog.Errorf("Failed to delete OVS port/interface %s: stderr: %s (%v)",
+				val, stderr, err)
+		}
+	}
+}
+
+// checkForStaleOVSRepresentorInterfaces checks for stale OVS ports backed by Repreresentor interfaces,
+// derrive iface-id from pod name and namespace then remove any interfaces assoicated with a sandbox that are
+// not scheduled to the node.
+func checkForStaleOVSRepresentorInterfaces(nodeName string, wf factory.ObjectCacheInterface) {
+	// Get all ovn-kuberntes Pod interfaces. these are OVS interfaces that have their external_ids:sandbox set.
+	out, stderr, err := util.RunOVSVsctl("--columns=name,external_ids", "--data=bare", "--no-headings",
+		"--format=csv", "find", "Interface", "external_ids:sandbox!=\"\"")
+	if err != nil {
+		klog.Errorf("failed to list ovn-k8s OVS interfaces:, stderr: %q, error: %v", stderr, err)
+		return
+	}
+
+	// parse this data into local struct
+	type interfaceInfo struct {
+		Name       string
+		Attributes map[string]string
+	}
+
+	var interfaceInfos []*interfaceInfo
```

Comment:
nit: you could allocate capacity here of len(lines) so the append doesnt have to keep recreating arrays

Reply:
done.